yeonjae/programmers-1/성격유형검사.js

```diff
@@ -0,0 +1,31 @@
+function solution(survey, choices) {
+  const result = {
+    R: 0,
+    T: 0,
+    C: 0,
+    F: 0,
+    J: 0,
+    M: 0,
+    A: 0,
+    N: 0,
+  };
+  let answer = "";
+
+  for (let i = 0; i < survey.length; i++) {
+    const [disagree, agree] = survey[i].split("");
+    if (choices[i] === 4) continue;
+    if (choices[i] < 4) result[disagree] += 4 - (choices[i] % 4);
+    else result[agree] += choices[i] % 4;
+  }
+
+  answer += result["R"] >= result["T"] ? "R" : "T";
+  answer += result["C"] >= result["F"] ? "C" : "F";
+  answer += result["J"] >= result["M"] ? "J" : "M";
+  answer += result["A"] >= result["N"] ? "A" : "N";
+
+  return answer;
+}
```

Comment:
```suggestion
function getResultPersonalityType(pointBoard) {
  const pointBoardEntries = Object.entries(pointBoard);
  let result = "";

  // 현재 타입과 다음 타입의 포인트를 비교해 최종 타입을 result에 적용
  for (let i = 0; i < pointBoardEntries.length; i += 2) {
    const [curType, curPoint] = pointBoardEntries[i];
    const [nextType, nextPoint] = pointBoardEntries[i + 1];
    const finalType = curPoint < nextPoint ? nextType : curType;

    result += finalType;
  }

  return result;
}

function solution(survey, choices) {
  const pointBoard = {
    R: 0,
    T: 0,
    C: 0,
    F: 0,
    J: 0,
    M: 0,
    A: 0,
    N: 0,
  };
  let answer = "";

  for (let i = 0; i < survey.length; i++) {
    const [disagree, agree] = survey[i].split("");
    if (choices[i] === 4) continue;
    if (choices[i] < 4) pointBoard[disagree] += 4 - (choices[i] % 4);
    else pointBoard[agree] += choices[i] % 4;
  }
    
  return getResultPersonalityType(pointBoard);
}
```
- 풀이 잘봤씁니다 ! 🙇‍♂️
- 아쉬운 것은 result와 answer의 의미가 비슷한 변수가 있다는 건데 result를 pointBoard 네이밍이 더 좋을 것 같다는 생각이 들었습니다.
- 코드를 개선해봤는데 더 복잡해보이지만 answer += ~ 의 반복되는 코드를 함수로 분리해서 적용한 내용입니다. 👍